src/sentry/rules/processing/processor.py

```diff
@@ -302,6 +334,39 @@ def apply_rule(self, rule: Rule, status: GroupRuleStatus) -> None:
                 )
                 return
 
+        predicate_func = get_match_function(condition_match)
+        if condition_list:
```

Comment:
I don't think we need this check. `any([]) == False` and `all([]) == True`, so we should just let an empty list go through the same `any/all` logic as a list with items

Reply:
I tried removing this but then it gets stuck on `if condition_match == "any" and not result:` - there isn't a result because there is no condition, and we get a default `condition_match` if one isn't provided.

Reply:
Right, I guess if we have filters and no conditions we don't want to fail on `ANY`...